### ISSUE TITLE
Analyzing unreliable CI 

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#12182]
I tried to get any clue on whats going on.

At this moment and over the last days the executors cgeo-1 and cgeo-5 have been online and serving the PR and commit jobs.

### Looking at cgeo-5:

It seems to fail most (but not all) builds with problems to install the APK on the emulator:
```
00:56:08  Installing APK 'cgeo-debug.apk' on 'android-30(AVD) - 11' for main:basic-debugUnable to install /home/jenkins/slave/workspace/cgeo CI - PR build/main/build/outputs/apk/basic/debug/cgeo-debug.apk
00:56:08  
00:56:08  com.android.ddmlib.InstallException: Unknown failure: Exception occurred while executing 'install':
00:56:08  java.lang.IllegalArgumentException: Error: Failed to parse APK file: /data/local/tmp/cgeo-debug.apk: Failed to parse /data/local/tmp/cgeo-debug.apk
00:56:08  at com.android.server.pm.PackageManagerShellCommand.setParamsSize(PackageManagerShellCommand.java:528)
00:56:08  at com.android.server.pm.PackageManagerShellCommand.doRunInstall(PackageManagerShellCommand.java:1282)
00:56:08  at com.android.server.pm.PackageManagerShellCommand.runInstall(PackageManagerShellCommand.java:1248)
00:56:08  at com.android.server.pm.PackageManagerShellCommand.onCommand(PackageManagerShellCommand.java:184)
00:56:08  at android.os.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:98)
00:56:08  at android.os.ShellCommand.exec(ShellCommand.java:44)
00:56:08  at com.android.server.pm.PackageManagerService.onShellCommand(PackageManagerService.java:21646)
00:56:08  at android.os.Binder.shellCommand(Binder.java:929)
00:56:08  at android.os.Binder.onTransact(Binder.java:813)
00:56:08  at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:4603)
00:56:08  at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:4305)
00:56:08  at android.os.Binder.execTransactInternal(Binder.java:1159)
00:56:08  at android.os.Binder.execTransact(Binder.java:1123)
00:56:08  Caused by: java.io.FileNotFoundException: AndroidManifest.xml
00:56:08  at android.content.res.ApkAssets.nativeOpenXml(Native Method)
00:56:08  at android.content.res.ApkAssets.openXml(ApkAssets.java:367)
00:56:08  at android.content.pm.parsing.ApkLiteParseUtils.parseApkLiteInner(ApkLiteParseUtils.java:245)
00:56:08  at android.content.pm.parsing.ApkLiteParseUtils.parseApkLite(ApkLiteParseUtils.java:226)
00:56:08  at com.android.server.pm.PackageManagerShellCommand.setParamsSize(PackageManagerShellCommand.java:522)
00:56:08  ... 12 more
```

I have no idea why this is happening?
Looks like either the APK is not present at the source or does not contain what should be in there (See: ```java.io.FileNotFoundException: AndroidManifest.xml```)

FYA:
I excluded cgeo-5 from job execution by removing its labels (```has-emulator has-emulator2```) in its preferences.

### Looking at cgeo-1:

This executors does at least seem not to show the installation problems of cgeo-5.
Looking at its build history it seems instead having sporadic problems, typically this failing initial test:
```
11:15:07  cgeo.geocaching.CgeoApplicationTest > testPreconditions[android-30(AVD) - 11] [31mFAILED [0m
11:15:07  	org.junit.ComparisonFailure: [User and password must be provided] expected:<[NO_ERROR]> but was:<[CONNECTION_FAILED]>
11:15:07  	at java.lang.reflect.Constructor.newInstance0(Native Method)
11:25:52  Build timed out (after 14 minutes). Marking the build as aborted.
11:25:52  FATAL: Caught exception while loading script 'cgeo build and test for new commits and pull requests v2'
```

Not sure, but looks like a problem to reach the server or problem with internet connection in general?
There are other sporadic failures on cgeo-1 as well, which I cannot judge as it also might be correct fails.

cgeo-5 is still excluded from building right now (see above).
Lets continue to use cgeo-1 to get a clear picture and try to fix that one first.

@mucek4 
Can you summarize here:
- Are cgeo-1 and cgeo-5 running on the same hardware?
- Could it be a internet connection or general network problem on cgeo-1?

**Comment by Lineflyer** [comment]
@mucek4 
Same builds (meaning tests for the same PR) which previously failed on cgeo-5 with the above mentioned problem (unable to install APK) pass on cgeo-1. This means cgeo-5 is somehow broken.
Can you try to re-instantiate it or freshly deploy it or make it a copy of cgeo-1 (whatever is possible or needed in the environment, I don't know the details).

Please lets align here on next steps to allow more analysis.

**Comment by kumy** [comment]
It took me 8 builds to get a running until the end test.

I'll build a runner on my side to compare the behavior.

**Comment by Lineflyer** [comment]
Thanks @kumy Any help appreciated.

The initial tests of your PR all run on cgeo-5 and failed with the above mentioned install error.
The green run was then on cgeo-1 after I took cgeo-5 out of the loop.

But right now cgeo-1 fails everything with the login fail described above.

**Comment by kumy** [comment]
Unfortunately we'll need @mucek4 here, JNLP port is closed, my executor cannot connect :/

**Comment by mucek4** [comment]
Executors 1 and 5 are running on different hardware. But share same internet line. But images are the same. Build from cgeo executor v4 branch.

For disabling a node, you have an option to disable it, right from nodes page. No need to remove tags.

@kumy do you have secrets for executors? If you have running docker with KVM support you can bring node 3 or 4 up.

Couple of days ago number 5 had the same problem and I have deleted workspace on it. Then I think it was fine.

**Comment by Lineflyer** [comment]
> For disabling a node, you have an option to disable it, right from nodes page. No need to remove tags.

Not sure how to do. I clicked on "Disconnect" in the node dialog. This disconnected it got reconnected after a few seconds.

**Comment by kumy** [comment]
@Lineflyer it's on the node page
![image](https://user-images.githubusercontent.com/176794/143733601-5f27b28e-ae2c-4b1d-8580-6c490833b16f.png)

> do you have secrets for executors?

@mucek4 … yes :) just need my ipv6 block to be allowed.And get the emulator starting in the container :/

**Comment by kumy** [comment]
- routing done on my side
- emulator up (`kumy-1`)
- First build https://ci.cgeo.org/job/cgeo-CI_PR-build/305/console

:crossed_fingers:

**Comment by kumy** [comment]
:champagne: 
```
Success Build #305 (Nov 28, 2021, 9:22:16 AM)
PR #12141: WIP: Testing GKv2 website o...
Took 9 min 21 sec on kumy-1
```

**Comment by kumy** [comment]
I have temporarily disabled `cgeo-1` and `cgeo-5`, I want to see how `kumy-1` behave.

**Comment by kumy** [comment]
Got the same error on 2nd build
```
09:37:48  com.android.ddmlib.InstallException: Unknown failure: Exception occurred while executing 'install':
09:37:48  java.lang.IllegalArgumentException: Error: Failed to parse APK file: /data/local/tmp/cgeo-debug.apk: Failed to parse /data/local/tmp/cgeo-debug.apk
```

**Comment by kumy** [comment]
This is the logcat error when it fails to parse the apk
```
11-28 08:47:48.285   540  1286 W PackageParsing: Failed to parse /data/local/tmp/cgeo-debug.apk
11-28 08:47:48.285   540  1286 W PackageParsing: java.io.FileNotFoundException: AndroidManifest.xml
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.content.res.ApkAssets.nativeOpenXml(Native Method)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.content.res.ApkAssets.openXml(ApkAssets.java:367)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.content.pm.parsing.ApkLiteParseUtils.parseApkLiteInner(ApkLiteParseUtils.java:245)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.content.pm.parsing.ApkLiteParseUtils.parseApkLite(ApkLiteParseUtils.java:226)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at com.android.server.pm.PackageManagerShellCommand.setParamsSize(PackageManagerShellCommand.java:522)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at com.android.server.pm.PackageManagerShellCommand.doRunInstall(PackageManagerShellCommand.java:1282)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at com.android.server.pm.PackageManagerShellCommand.runInstall(PackageManagerShellCommand.java:1248)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at com.android.server.pm.PackageManagerShellCommand.onCommand(PackageManagerShellCommand.java:184)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.os.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:98)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.os.ShellCommand.exec(ShellCommand.java:44)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at com.android.server.pm.PackageManagerService.onShellCommand(PackageManagerService.java:21646)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.os.Binder.shellCommand(Binder.java:929)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.os.Binder.onTransact(Binder.java:813)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:4603)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:4305)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.os.Binder.execTransactInternal(Binder.java:1159)
11-28 08:47:48.285   540  1286 W PackageParsing: 	at android.os.Binder.execTransact(Binder.java:1123)
```

**Comment by kumy** [comment]
Analyzing `apk` content, `AndroidManifest.xml` is indeed missing
```
jenkins@1584bb76f547:~/BAK3/outputs/apk/basic/debug$ unzip -t cgeo-debug.apk |grep AndroidManifest.xml
```

```
enkins@1584bb76f547:~/BAK3/outputs/apk/basic/debug$ unzip -t cgeo-debug.apk |grep xml
    testing: META-INF/services/com.fasterxml.jackson.core.JsonFactory   OK
    testing: META-INF/services/com.fasterxml.jackson.core.ObjectCodec   OK
    testing: assets/mapsforge/default.xml   OK
    testing: assets/mapsforge/osmarender.xml   OK
    testing: org/jacoco/agent/rt/internal_8ff85ea/asm/xml/asm-xml.dtd   OK
    testing: pois.xsd3b.xml           OK

No errors detected in compressed data of cgeo-debug.apk.
```

**Comment by kumy** [comment]
This one is passing https://ci.cgeo.org/job/cgeo-CI_PR-build/311/console
It downloaded an earlier version of `gradle-6.8.3`, while other seems to use `6.9.1`

But no :/
https://ci.cgeo.org/job/cgeo-CI_PR-build/311/ -> same PR working
https://ci.cgeo.org/job/cgeo-CI_PR-build/314/ -> same PR failing

**Comment by kumy** [comment]
Doing some manual deeper tests, all runners are currently disabled.

**Comment by kumy** [comment]
@Lineflyer 

After hours with manual builds git bisecting, scratching and banging my head on the table, I'm glad to announce I've finally get something quite stable, at least for the last 8 builds (starting from `#3​28`)…

Don't ask me why, but after having decoupled the old build line to small steps, it seems to work more reliably:
```
# From
./gradlew --no-daemon --scan clean testDebug createBasicDebugCoverageReport checkstyle lintBasicDebug -Pandroid.testInstrumentationRunnerArguments.notAnnotation=cgeo.geocaching.test.NotForIntegrationTests

---

# To
./gradlew --no-daemon --scan  clean
rm -fr build build-cache main/build || true

./gradlew --no-daemon --scan  testDebug -Pandroid.testInstrumentationRunnerArguments.notAnnotation=cgeo.geocaching.test.NotForIntegrationTests
./gradlew --no-daemon --scan  checkstyle
./gradlew --no-daemon --scan  lintBasicDebug
```

From what I saw, the main problem was the apk generation. The resulting file is normally ~17M, but seen it at 2.2M, 10M when it fails. The `AndroidManifest.xml`, and all `res/` folder is missing.

Maybe some graddle guru can figure out why this little guy has problem :)

"Required" changes are applied to script https://ci.cgeo.org/configfiles/editConfig?id=0591d16e-bf19-43f3-9421-af26d4d45276, Also added some `Failure causes` in `Failure Cause Management`

**Executors `cgeo-1` `cgeo-2` are back online (tag re-added)** + I let `kumy-1` up some days.

I really hope it will stay stable :crossed_fingers: :crossed_fingers: :crossed_fingers: :crossed_fingers: :crossed_fingers:

**Comment by Lineflyer** [comment]
@kumy 

Now the APK problem is back at least on job https://ci.cgeo.org/job/cgeo-CI_commit-build/ (did not check others).

**Comment by Lineflyer** [comment]
Here is also a fail on "your" slave: https://ci.cgeo.org/job/cgeo-CI_commit-build/108/console

**Comment by kumy** [comment]
Only the other job (PR something) is configured to use the v2 script ;)
On the PR job side it seems to be reliable.

**Comment by kumy** [comment]
> Here is also a fail on "your" slave: https://ci.cgeo.org/job/cgeo-CI_commit-build/108/console

From what I saw, the problem has nothing to do with executor themselves. The problem is on gradle side parallelising too much and start destroying file before another task ends. Maybe "too much cores" available on new executors :D.

I'll switch other jobs to the new script in a moment. ;)

**Comment by kumy** [comment]
Done on `cgeo-CI_commit-build` job

**Comment by kumy** [comment]
They are now failing by a 3min timeout.
I'll update the conf from the dynamic thing 
![image](https://user-images.githubusercontent.com/176794/144570267-8e94dba8-3d66-473a-8f02-170043c4fe53.png)

to 20 min...

**Comment by Lineflyer** [comment]
Thanks @kumy, looks OK now.

I just noted another (seemingly unrelated) problem:

The feedback on commit build (green tick mark behind commit) stopped working around November, 20th.

Looking at a [recent commit build](https://ci.cgeo.org/job/cgeo-CI_commit-build/145/console) it claims to have succesfully submitted it to Github:

21:16:00  [Set GitHub commit status (universal)] SUCCESS on repos [] (sha:effae45) with context:cgeo-CI_commit-build

The commit however has no tick mark:
![image](https://user-images.githubusercontent.com/949669/144916587-cf7a861d-8ddb-41ce-b9e8-749f3acab02b.png)

@kumy Did you change any of the post build actions?

**Comment by kumy** [comment]
> Thanks @kumy, looks OK now.

Glad to have been helpful. All the changes I made are referenced in this issue history.
no idea about the feedback in commit history :/

**Comment by Lineflyer** [comment]
Found something worth a try. Working on it.

**Comment by kumy** [comment]
`kumy-1` executor is now down. :)

**Comment by Lineflyer** [comment]
Fixed.
It was a aftermath of #12029:
While it was fixed for the checkout phase by adding authentication it seems the Github plugin on Jenkins itself needs a dedicated authentication setup. I added the same entry of the credential store to be used in the Github plugin configuration and re-run the build. Now the status is shown again:

![image](https://user-images.githubusercontent.com/949669/144919720-da30c76f-d80d-4981-8290-c0cef86c4be6.png)


Closing here.
Instability is fixed thanks to @kumy and now all integration parts should also be running again.
